src/program/graph.rs

```diff
@@ -233,24 +301,47 @@ impl InstructionBlock {
                     Ok(())
                 }
                 InstructionRole::RFControl => {
-                    let used_frames = program
+                    let used_frames: HashSet<&FrameIdentifier> = program
```

Comment:
`get_frames_for_instruction` returns an `Option<Vec<<&FrameIdentifier>>`, but its only uses always convert that to a `HashSet<&FrameIdentifier>`. Would it make sense to DRY this out and change the function definition?

Reply:
Yes - although that's a breaking change - _although_ that's fine these days.